src/components/CurrencyPicker.tsx

```diff
@@ -1,89 +1,105 @@
+
+    /** Should show the full page offline view (whenever the user is offline) */
+    shouldShowFullPageOfflineView?: boolean;
 };
-function CurrencyPicker({selectedCurrency, label = '', errorText = '', value, onInputChange, furtherDetails}: ValuePickerProps & CurrencyPickerProps, forwardedRef: ForwardedRef<View>) {
-    const StyleUtils = useStyleUtils();
-    const styles = useThemeStyles();
-    const [isPickerVisible, setIsPickerVisible] = useState(false);
-    const {translate} = useLocalize();
 
-    const showPickerModal = () => {
-        setIsPickerVisible(true);
-    };
+function CurrencyPicker({label, value, errorText, headerContent, excludeCurrencies, interactive, shouldShowFullPageOfflineView = false, onInputChange = () => {}}: CurrencyPickerProps) {
+    const {translate} = useLocalize();
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+    const styles = useThemeStyles();
 
     const hidePickerModal = () => {
         setIsPickerVisible(false);
     };
 
-    const updateInput = (item: ValuePickerItem) => {
-        if (item.value !== selectedCurrency) {
-            onInputChange?.(item.value);
-        }
+    const updateInput = (item: CurrencyListItem) => {
+        onInputChange?.(item.currencyCode);
         hidePickerModal();
     };
 
-    const descStyle = !selectedCurrency || selectedCurrency.length === 0 ? StyleUtils.getFontSizeStyle(variables.fontSizeLabel) : null;
+    const BlockingComponent = shouldShowFullPageOfflineView ? FullPageOfflineBlockingView : Fragment;
 
     return (
-        <View>
+        <>
             <MenuItemWithTopDescription
-                ref={forwardedRef}
                 shouldShowRightIcon
-                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-                title={value || ''}
-                descriptionTextStyle={descStyle}
+                title={value ? `${value} - ${getCurrencySymbol(value)}` : undefined}
                 description={label}
-                onPress={showPickerModal}
-                furtherDetails={furtherDetails}
+                onPress={() => setIsPickerVisible(true)}
                 brickRoadIndicator={errorText ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                 errorText={errorText}
+                interactive={interactive}
             />
-
             <Modal
                 type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
                 isVisible={isPickerVisible}
                 onClose={hidePickerModal}
                 onModalHide={hidePickerModal}
                 hideModalContentWhileAnimating
                 useNativeDriver
-                onBackdropPress={hidePickerModal}
+                onBackdropPress={Navigation.dismissModal}
             >
                 <ScreenWrapper
-                    style={styles.pb0}
+                    style={[styles.pb0]}
                     includePaddingTop={false}
-                    includeSafeAreaPaddingBottom={false}
```

Comment:
coming from https://github.com/Expensify/App/issues/56697

we caused an issue where offline message is behind the keyboard